Allow usage of unauthenticated PuppetDB over SSL

cassianoleal opened the pull request:
On some occasions the user might have a PuppetDB installation serving the API from an SSL-enabled server with no authentication mechanism. In these cases the pem settings are not necessary and should be avoided.

Poil commented:
+1

queeno commented:
+1

puppetcla commented:
CLA signed by all contributors.

cassianoleal commented:
Anyone from @voxpupuli could have a look at this, perhaps add a comment if you're not happy with the change?

dhollinger commented:
@cassianoleal Sorry for the delay on this.

Could you rebase your PR off master, then I'll have a look at it.

cassianoleal commented:
@dhollinger I have rebased and updated the specs. All tests are passing.

I've stopped using Puppet altogether and don't have a real PuppetDB installation to test this against but other people seem to want this capability as well so I made the effort.

If there's anything to improve please let me know.

dhollinger commented:
@cassianoleal Thanks!

I'm going to bring this topic up in the `voxpupuli` IRC channel and see what the community thinks. This requires more discussion before any one person merges it
